### PR TITLE
Add how-to page to backup and restore Keycloak

### PR DESCRIPTION
**Pull request by ccremer**
Adds a tutorial that explains how a Backup and Restore of the built-in database is done with K8up.

Depends on #106 

## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.

### CODE REVIEW

**Comment by glrf** [comment]
@ccremer  I somehow cannot request a review from you, but feel free to give input.

File: docs/modules/ROOT/pages/how-tos/restore.adoc [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+----
+
+[NOTE]
+====
+This guide assumes that you have direct access to the S3 bucket holding the backup.
+If the access is restricted to the Kubernetes cluster, you will need to adapt these steps.
+You could:
+
+. Perform the commands in a container running on the cluster
+. Restore the database backup to a PVC and copy if over
```
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/restore.adoc:99]:
```suggestion
. Restore the database backup to a PVC and copy it over
```

File: docs/modules/ROOT/pages/how-tos/restore.adoc [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+[NOTE]
+====
+You can only restore a database that has been backed up.
+Please refer to the xref:how-tos/use-built-in-db.adoc#_enable_backups[built-in database] setup guide on how to enable backups.
+====
+
+====
+Requirements
+
+* `kubctl` With access to the cluster running Keycloak
```
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/restore.adoc:20]:
```suggestion
* `kubectl` With access to the cluster running Keycloak
```

**Review by simu — APPROVED** [review]
LGTM overall

File: docs/modules/ROOT/pages/how-tos/restore.adoc [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+* `base64`
+* https://restic.net/[`restic`]
+* (optional) https://stedolan.github.io/jq/[`jq`]
```
Comment (ccremer) [inline docs/modules/ROOT/pages/how-tos/restore.adoc:23]:
```suggestion
* (optional) https://stedolan.github.io/jq/[`jq`]
* `pwgen`
* `vault`
```

File: docs/modules/ROOT/pages/how-tos/restore.adoc [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+. Load the backup and restore it
++
+[source,bash]
+----
+export POD=keycloak-postgresql-0
+
+restic dump "${SNAPSHOT_ID}" /$NAMESPACE-keycloak-postgresql.sql \
+  | kubectl -n $NAMESPACE exec -i $POD \
+  -- bash -c 'PGPASSWORD="${POSTGRES_PASSWORD}" psql --set ON_ERROR_STOP=on -U "${POSTGRES_USER}" ${POSTGRES_DB}'
```
Comment (ccremer) [inline docs/modules/ROOT/pages/how-tos/restore.adoc:89]: Does regular `sh` also work? The user might use an image that doesn't have bash installed (e.g. Alpine variant)
```suggestion
  -- sh -c 'PGPASSWORD="${POSTGRES_PASSWORD}" psql --set ON_ERROR_STOP=on -U "${POSTGRES_USER}" ${POSTGRES_DB}'
```

File: docs/modules/ROOT/pages/how-tos/use-built-in-db.adoc [hunk 5]
```diff
@@ -36,3 +36,59 @@ vault kv put -cas=0 "${key}" admin-password=$(pwgen -s 32 1) db-password=$(pwgen
...
+s3_access_key=YOUR_ACCESS_KEY
+s3_secret_key=YOUR_SECRET_KEY
+key="clusters/kv/${TENANT_ID}/${CLUSTER_ID}/keycloak"
+
+vault kv put -cas=0 "${key}" k8up-s3-accesskey=${s3_access_key} k8up-s3-secretkey=${s3_secret_key}
+----
++
+[TIP]
+====
+On most clusters you should be able to reuse the global backup credentials setup during cluster creation.
```
Comment (ccremer) [inline docs/modules/ROOT/pages/how-tos/use-built-in-db.adoc:81]:
```suggestion
On most clusters you should be able to reuse the global backup credentials set up during cluster creation.
```

File: docs/modules/ROOT/pages/how-tos/restore.adoc [hunk 6]
```diff
@@ -0,0 +1,104 @@
+= Restore Keycloak
```
Comment (ccremer) [inline docs/modules/ROOT/pages/how-tos/restore.adoc:1]:
```suggestion
= Restore Keycloak from a Backup
```

File: docs/modules/ROOT/partials/nav.adoc [hunk 7]
```diff
@@ -12,6 +12,7 @@
 * xref:how-tos/configure-ingress.adoc[Configure Keycloak ingress]
 * xref:how-tos/custom-theme.adoc[Configure custom theme]
 * xref:how-tos/change-passwords.adoc[Change passwords]
+* xref:how-tos/restore.adoc[Restore a backup]
```
Comment (ccremer) [inline docs/modules/ROOT/partials/nav.adoc:15]: I think titles should somewhat match the menu item, if not too long
```suggestion
* xref:how-tos/restore.adoc[Restore from a Backup]
```

File: docs/modules/ROOT/pages/how-tos/restore.adoc (hunk 4 above)
Comment (glrf) [reply docs/modules/ROOT/pages/how-tos/restore.adoc:89]: Should work I think. I'll double-check

**Pull request merged**